### DISCUSSION TITLE
Does blink.cmp work fast for you in non-empty files?

### DISCUSSION THREAD

**Discussion by skoch13** [LazyVim/LazyVim#5097] [General]
Recently I was using `magazine.nvim` and it was decent, but after switching to `blink.cmp` I observe slow completion suggestions when file is not empty. E.g. `typescriptreact`

https://github.com/user-attachments/assets/64737556-9b56-4dcc-927a-0deb4f555992

Is it possibly related to buffer words blink trying to utilize?

**Comment by folke** [comment]
fyi, I just pushed some updates related to blink and ai providers.
AI providers now get a score boost of `100` and I've also enabled `async=true`, which means
the blink menu will already show while ai results are still fetched.

The delay you noticed was blink waiting for copilot results.
Adding the `async=true` fixes this.

Do let me know if it's not better for you.

**Reply by skoch13** [reply]
thank you. I added changes from the commit you mentioned to the configuration and it does make changes(put copilot suggestions on top), but delay is still there 

https://github.com/user-attachments/assets/1d6ee75a-872b-48b4-8688-15b461043826

**Reply by folke** [reply]
Is that once your LSP is fully loaded and ready?

**Reply by folke** [reply]
Just tested with a `js` file and everyting is instant. No delay at all.

**Reply by skoch13** [reply]
yes, LSP is fully ready. I even tried creating  empty `tsx` file and result is almost identical 

https://github.com/user-attachments/assets/e979bdeb-c2ef-44b9-b294-1de05204e74f

**Reply by folke** [reply]
Try adding `opts.source.providers.lsp.async = true`.
Might be just the LSP that's slow and nvim-cmp probably shows directly while lsp results are loading,
while blink by default will wait till it has the results (unless you set `async = true`)

**Comment by folke** [comment]
This is how it looks for me:

https://github.com/user-attachments/assets/9cc531bd-3f17-4a7f-b0b7-afe691b22822

Did you enable any additional sources? extras?

**Reply by skoch13** [reply]
> Did you enable any additional sources? extras?

No other sources, there are some extras, but they're not related to completion.

I checked another TS project, and it works so much better. So I guess the root cause might be LSP being slow in the monorepo in general, but that doesn't happen in VSCode and as far as I understand both of them use `vtsls`.

> Try adding opts.source.providers.lsp.async = true.

Tried it as well, but it didn't make much difference. I guess I need a way to diagnose completion calculations bottleneck to be more precise. I'll ask for help in the respective repo.

**Comment by guillenotfound** [comment]
Uhhm for me there's something going on with `blink.cmp` too.

When there's a suggestion (LSP) and I hit either `ctrl+y` or `enter` to accept it, it will have a delay before the selected suggestion appears in the buffer.

Enabling the extra `nvim-cmp` seems to be faster at first sight. What information can I provide such that I can help finding the root cause?

Also after checking the docs I think it's `opts.sources` (notice the trailing `s`) https://github.com/LazyVim/LazyVim/discussions/5097#discussioncomment-11573928

**Reply by guillenotfound** [reply]
Looks like it might be related to: https://github.com/Saghen/blink.cmp/issues/627

**Reply by skoch13** [reply]
I switched back to nvim-cmp(magazine.nvim to be precise) because for now blink is barely usable for me unfortunately

**Comment by skoch13** [comment]
Latest blink release addressed all of my concerns. Works like magic 🥇

**Reply by skoch13** [reply]
...aaand after yet another update I had to switch back to magazine.nvim

**Reply by evertonse** [reply]
not again
